File: Backend/src/main/java/com/luckyseven/backend/domain/email/entity/EmailVerificationToken.java
package com.luckyseven.backend.domain.email.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "email_verfication_tokens",indexes = @Index(columnList = "token",unique = true))
public class EmailVerificationToken {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false , unique = true , length = 36)
  private String token;

  @Column(nullable = false)
  private String email;

  @Column(nullable = false)
  private LocalDateTime expireTime;

}

File: Backend/src/main/kotlin/com/luckyseven/backend/domain/email/entity/EmailVerificationToken.kt
package com.luckyseven.backend.domain.email.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(
    name = "email_verification_tokens",
    indexes = [Index(columnList = "token", unique = true)]
)
 class EmailVerificationToken(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    
    @Column(nullable = false, unique = true, length = 36)
    val token: String,
    
    @Column(nullable = false)
    val email: String,
    
    @Column(nullable = false)
    val expireTime: LocalDateTime
) 